Add `turbo_frame` option to `redirect_to` action

adrianthedev opened the pull request:
Adds the `frame` option to the `redirect_to` action.
Super useful for when you'd like to perform an update and a redirect using turbo streams.

I think the `frame` keyword argument is better suited for this rather than a positional argument, but it's up to you to set the API.

Thanks!

<img width="928" alt="CleanShot 2023-08-22 at 03 11 09@2x" src="https://github.com/marcoroth/turbo_power/assets/1334409/7fef66c4-07a5-4f57-8829-291822078169">

marcoroth commented:
Hey @adrianthedev, nice one, thank you!

Regarding the API: `turbo_power-rails` supports both keyword and positional arguments thanks to https://github.com/marcoroth/turbo_power-rails/pull/26.

So I think to keep it consistent we can add `frame` (or even `turbo_frame` to align with `turbo_action`) as a third optional positional argument and also have it being supported via keyword arguments.

adrianthedev commented:
Gotcha! Let me put the finishing touches and add some tests over the coming days.

marcoroth commented:
> Gotcha! Let me put the finishing touches and add some tests over the coming days.

Awesome, let me know if you need any guidance or when you get stuck somewhere 🙌🏼

adrianthedev commented:
I addressed your feedback.

Not sure why [this](https://github.com/marcoroth/turbo_power/pull/79/files#diff-5b3fd4a00a8fe358a1587e7d27fbdce4e8d29b819505ab41cb347303a6ee9dcfR149) test is failing. 
I followed your lead when I created it.
Could you please help me out with that?

adrianthedev commented on src/actions/turbo.ts:
```diff
@@ -5,10 +5,18 @@ import Proxy from "../proxy"
   const turboAction = (this.getAttribute("turbo-action") || "advance") as Action
+  const turboFrame = this.getAttribute("turbo-frame") as Action
   const turbo = this.getAttribute("turbo") !== "false"
+  const args: Partial<{ action: Action; frame?: string }> = {
+    action: turboAction,
+  }
```
I figured there must be a type already. I couldn't find it.

marcoroth commented on test/turbo/redirect_to.test.js:
```diff
@@ -145,4 +145,18 @@ describe("redirect_to", () => {
+    it("renders the frame attribute", async () => {
+      sinon.replace(window, "Turbo", { visit: sinon.fake(), frame: "modals" })
+
+      await executeStream(`<turbo-stream action="redirect_to" frame="modals" url="http://localhost:8080"></turbo-stream>`)
```
The test is failing because it expects the `frame` attribute to be `turbo-frame`. I know it's confusing, because the `action` attribute right next to it is not the `turbo-action` attribute, but the `action` from the Turbo Stream which determines which actually it actually is.

```suggestion
      await executeStream(`<turbo-stream action="redirect_to" turbo-frame="modals" url="http://localhost:8080"></turbo-stream>`)
```

adrianthedev commented on src/actions/turbo.ts:
```diff
@@ -5,10 +5,18 @@ import Proxy from "../proxy"
 export function redirect_to(this: StreamElement) {
   const url = this.getAttribute("url") || "/"
   const turboAction = (this.getAttribute("turbo-action") || "advance") as Action
+  const turboFrame = this.getAttribute("turbo-frame") as Action
```
You're right!

marcoroth commented:
Thank you @adrianthedev!
